pydarn/plotting/fan.py

```diff
@@ -172,31 +177,70 @@ def plot_fan(cls, dmap_data: List[dict], ax=None,
                     found_match = True
                     break
             # handle datetimes out of bounds
-            if found_match == False:
+            if found_match is False:
                 raise plot_exceptions.IncorrectDateError(rec_time,
                                                          scan_time)
         # Locate scan in loaded data
         plot_beams = np.where(beam_scan == scan_index)
-
         # Time for coordinate conversion
         if not scan_time:
-        	date = time2datetime(dmap_data[plot_beams[0][0]])
+            date = time2datetime(dmap_data[plot_beams[0][0]])
         else:
-        	date = scan_time
+            date = scan_time
 
         # Plot FOV outline
-        if ranges is None:
-            ranges = [0, dmap_data[0]['nrang']]
+        stid = dmap_data[0]['stid']
+        if ranges == [] or ranges is None:
+            try:
+                # If not given, get ranges from data file
+                ranges = [0, dmap_data[0]['nrang']]
+            except KeyError:
+                # Otherwise, default to [0,75]
+                ranges = [0, SuperDARNRadars.radars[stid].range_gate_45]
 
-        beam_corners_aacgm_lats, beam_corners_aacgm_lons, thetas, rs, ax = \
-            cls.plot_fov(dmap_data[0]['stid'], date, ax=ax, **kwargs)
+        # Get rsep and frang from data unless not there then take defaults
+        # of 180 km for frang and 45 km for rsep as these are most commonly
+        # used
+        try:
+            frang = dmap_data[0]['frang']
+        except KeyError:
+            frang = 180
 
-        fan_shape = beam_corners_aacgm_lons.shape
+        try:
+            rsep = dmap_data[0]['rsep']
+        except KeyError:
+            rsep = 45
+
+        if coords != Coords.GEOGRAPHIC and projs == Projs.GEO:
+            raise plot_exceptions.NotImplemented("AACGM coordinates are"\
+                                                 " not implemented for "\
+                                                 " geographic projections right"\
+                                                 " now, if you would like to"\
+                                                 " see it sooner please help"\
+                                                 " out at "\
+                                                 "https://github.com"\
+                                                 "/SuperDARN/pyDARN")
+
+        beam_corners_lats, beam_corners_lons =\
```

Comment:
Sorry for a single comment here. I realize that the return data beam_corners_lons from the fan.py is in fact the shifted mlt if the coords=AACGM_MLT. But in the parameter description, it says "AACGMv2 longitudes" is returned. And I would prefer to return the actual mlt in hours instead of the shifted mlt from aacgm_MLT_coordinates. Any comment on why the shifted mlt is returned?

Reply:
I have amended the parameter description to say that the return values are dependent on the chosen or default coords value chosen, which I believe is what we discussed in meeting last week. I personally can't comment on why it's in shifted MLT, but as it stands it is working, the solution may be to have the lines of code that do the shifting in here rather than in coordinates.py? Otherwise, I think it's a little too close to the hdw deadline to refactor this right now and have time for people to test, so it might be best to make an issue and we can discuss it for a future minor release?